plugins/woocommerce-blocks/assets/js/base/context/hooks/cart/use-store-cart.ts

```diff
@@ -230,9 +252,9 @@ export const useStoreCart = (
 				cartTotals,
 				cartIsLoading,
 				cartErrors,
-				billingData: emptyHiddenAddressFields( billingAddress ),
-				billingAddress: emptyHiddenAddressFields( billingAddress ),
-				shippingAddress: emptyHiddenAddressFields( shippingAddress ),
+				billingData: billingAddressRef.current,
+				billingAddress: billingAddressRef.current,
+				shippingAddress: shippingAddressRef.current,
```

Comment:
I introduced refs so we could compare with the previous value since these are objects.